### PR TITLE
[FIX] account: Fix compute of epd when invoice is draft

### PR DESCRIPTION
**Pull request by smetl**
When the invoice is a New record and thus, not yet stored inside the database, the totals like amount_untaxed/amount_tax/amount_total are set to 0.0 because line_ids is empty.
About the lines, balance/amount_currency are also set to 0.0.
This means we can not rely to any of these fields in any case when computing the values for a New record.

To reproduce, configure the early payment discount computation to "Always upon invoice", create a new invoice and set the demo terms 30 Net 2/7.
For a customer invoice, the tax amount will not be correct until the save.
For a vendor bill, the tax amount will be erased by the quick edit mode and the correct value will never be set.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by casp-odoo** [comment]
@yosa-odoo :point_up:

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -946,18 +946,33 @@ def _compute_payment_state(self):
...
+                            untaxed_amount_currency += line.price_subtotal
+                            for tax_result in (line.compute_all_tax or {}).values():
+                                tax_amount_currency += -sign * tax_result.get('amount_currency', 0.0)
+                        untaxed_amount = untaxed_amount_currency
+                        tax_amount = tax_amount_currency
+                    else:
+                        tax_amount_currency = invoice.amount_tax * sign
+                        tax_amount = invoice.amount_tax_signed
+                        untaxed_amount_currency = invoice.amount_untaxed * sign
+                        untaxed_amount = invoice.amount_untaxed_signed
```
Comment (william-andre) [inline addons/account/models/account_move.py:968]: Is it not possible to do this in `_compute_amount`?

**Review by smetl — COMMENTED** [review]
@william-andre test added

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**